kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/spec/tempfile.kt

```diff
@@ -2,13 +2,16 @@ package io.kotest.engine.spec
 
 import io.kotest.core.TestConfiguration
 import java.io.File
+import kotlin.io.path.deleteExisting
 
 fun TestConfiguration.tempfile(prefix: String? = null, suffix: String? = null): File {
-  
    val file = kotlin.io.path.createTempFile(prefix ?: javaClass.name, suffix).toFile()
    afterSpec {
-      if (!file.delete())
+      runCatching {
+         file.toPath().deleteExisting()
+      }.onFailure {
```

Comment:
do we need any tests changed/added to go with this code changes?

Reply:
It's difficult to do, as at least the JVM platform does not support creating such files with unmappable characters to test their deletion...